lib/sigstore/oidc.rb

```diff
@@ -20,7 +20,8 @@ module OIDC
       "https://accounts.google.com" => "email",
       "https://oauth2.sigstore.dev/auth" => "email",
       "https://oauth2.sigstage.dev/auth" => "email",
-      "https://token.actions.githubusercontent.com" => "job_workflow_ref"
+      "https://token.actions.githubusercontent.com" => "job_workflow_ref",
+      "https://agent.buildkite.com" => "pipeline_slug"
```

Comment:
we actually need *two* claims to generate the full identity for a Buildkite fulcio certificate. I do that further down, which is a bit messy